### PR TITLE
Rewrote breaking change documentation to be more compact and task-based

### PR DESCRIPTION
**Pull request by melinath**
Main changes are to https://googlecloudplatform.github.io/magic-modules/develop/make-a-breaking-change/ and https://googlecloudplatform.github.io/magic-modules/reference/breaking-change-detector/

localhost_1313_magic-modules_develop_breaking-changes_

localhost_1313_magic-modules_develop_make-a-breaking-change_ (1)

```release-note:none

```

### CODE REVIEW

File: docs/content/develop/make-a-breaking-change.md [hunk 1]
```diff
@@ -1,211 +1,213 @@
...
+{{< tab "MMv1" >}}
+1. Set `docs.warning` and `custom_code.resource_definition` on the resource. For example:
+
+   ```yaml
+   docs: !ruby/object:Provider::Terraform::Docs
+     warning: |
+       google_RESOURCE_NAME is deprecated and will be removed in the next major release
+       of the provider. Use google_OTHER_RESOURCE_NAME instead.
+   custom_code: !ruby/object:Provider::Terraform::CustomCode
+     resource_definition: templates/terraform/resource_definition/RESOURCE_NAME_deprecation.go.erb
```
Comment (c2thorn) [inline docs/content/develop/make-a-breaking-change.md:131]: Is the resourc_definition necessary? What does it do?

Comment (c2thorn) [reply docs/content/develop/make-a-breaking-change.md:131]: Oh, I see. I think recently we've added `deprecation_message` as an attribute. That should replace this step.

Comment (melinath) [reply docs/content/develop/make-a-breaking-change.md:131]: I don't think we support it yet so I've created a PR to add support at https://github.com/GoogleCloudPlatform/magic-modules/pull/8696. I'll update this PR accordingly.

Comment (melinath) [reply docs/content/develop/make-a-breaking-change.md]: done in fd875dd

**Review by c2thorn — APPROVED** [review]
Seems good based on pending deprecation_message PR

**Pull request merged**